test/cluster/test_cdc_with_tablets.py

```diff
@@ -292,3 +292,51 @@ async def test_cdc_colocation(manager: ManagerClient):
                     f"Stream {stream_id} partitions {partitions} are not fully contained in inaccessible partitions"
 
         await manager.server_start(servers[0].server_id)
+
+# Test garbage collection of CDC streams.
+# Create a CDC table with short TTL, then split the tablets to create a new stream set,
+# and wait until the old streams are garbage collected and we have a single stream set again.
+# Verify the remaining stream set is equal to the most recent stream set.
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_cdc_stream_garbage_collection(manager: ManagerClient):
+    cfg = { 'tablet_load_stats_refresh_interval_in_seconds': 1, 'error_injections_at_startup': ['short_cdc_streams_gc_refresh_interval' ] }
+    servers = await manager.servers_add(1, config=cfg)
+    cql = manager.get_cql()
+    async with new_test_keyspace(manager, "WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 2}") as ks:
+        await cql.run_async(f"CREATE TABLE {ks}.test (pk int PRIMARY KEY, v int) WITH cdc={{'enabled': true, 'ttl': 5}}")
+
+        rows = await cql.run_async(f"SELECT * FROM system.cdc_timestamps WHERE keyspace_name='{ks}' AND table_name='test'")
+        assert len(rows) == 1
+
+        await cql.run_async(f"ALTER TABLE {ks}.test_scylla_cdc_log WITH tablets = {{'min_tablet_count': 4}};")
+
+        async def new_stream_timestamp_created():
+            rows = await cql.run_async(f"SELECT * FROM system.cdc_timestamps WHERE keyspace_name='{ks}' AND table_name='test'")
+            if len(rows) > 1:
+                return True
+        await wait_for(new_stream_timestamp_created, time.time() + 60)
+
+        ts_before = await cql.run_async(f"SELECT * FROM system.cdc_timestamps WHERE keyspace_name='{ks}' AND table_name='test'")
+        streams_before = await cql.run_async(f"SELECT * FROM system.cdc_streams WHERE keyspace_name='{ks}' AND table_name='test'")
```

Comment:
If you do want to observe two streams, then I suggest changing ttl to a large value (e.g. 1h, 1 day, etc), then changing it to a low value before executing the `streams_garbage_collected` wait.